### PR TITLE
Automatic insights

### PR DESCRIPTION
**Pull request by deniskaber**
Relates to https://github.com/metabase/metabase/issues/26836

### Description

- The `X-ray` and `Compare to the rest` actions should be under a new menu option, `Automatic insights…`
- Clicking this option should then show the two options, `X-ray` and `Compare to the rest`

### Demo

https://user-images.githubusercontent.com/7983744/235778006-0332c381-b9b4-441b-bfa8-fd203546eca6.mov

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30370)

### CODE REVIEW

File: frontend/src/metabase/visualizations/components/ChartClickActions/utils.ts [hunk 1]
```diff
@@ -29,9 +29,6 @@ export const SECTIONS: Record<string, Section> = {
   standalone_filter: {
     icon: "filter",
   },
-  filter: {
```
Comment (uladzimirdev) [inline frontend/src/metabase/visualizations/components/ChartClickActions/utils.ts:32]: any reason to move it below?

File: frontend/src/metabase/modes/components/drill/common/DrillActionsListPopover.styled.tsx [hunk 2]
```diff
@@ -5,9 +5,9 @@ import Button from "metabase/core/components/Button";
 import BreakoutPopover from "metabase/query_builder/components/BreakoutPopover";
 
 export const Container = styled.div`
-  min-width: 210px;
+  min-width: 256px;
```
Comment (uladzimirdev) [inline frontend/src/metabase/modes/components/drill/common/DrillActionsListPopover.styled.tsx:8]: magic numbers?

**Review by uladzimirdev — COMMENTED** [review]
LGTM

**Review by ranquild — COMMENTED** [review]
Looks good. I think we can improve the case with a single action in the drill down component. Wdyt?

> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/30370#-:-NUX-9JnBQC3JgDxoPCf:bilx2ue)* status: 0 of 13 files reviewed, 3 unresolved discussions (waiting on @deniskaber)

___
*[`frontend/src/metabase/modes/components/drill/AutomaticInsightsDrill.tsx` line 72 at r1](https://reviewable.io/reviews/metabase/metabase/30370#-NUWzmHB-_1DQdD8qc-r:-NUWzmHB-_1DQdD8qc-s:btfl7k2) ([raw file](https://github.com/metabase/metabase/blob/99fb47d9e0d149119b3c5e5bd49bf1fd5757be8b/frontend/src/metabase/modes/components/drill/AutomaticInsightsDrill.tsx#L72)):*
> ```XML
>     const dispatch = useDispatch();
> 
>     if (drillOptions.length === 1) {
> ```

Instead of doing it like this, could we return a single click action with the title of "Automatic insights" and the url of the first drill in this case? So we won't need to call dispatch during the render of the popover

**Review by deniskaber — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/30370)* status: 0 of 15 files reviewed, 3 unresolved discussions (waiting on @deniskaber, @ranquild, and @uladzimirdev)

___
*[`frontend/src/metabase/modes/components/drill/AutomaticInsightsDrill.tsx` line 72 at r1](https://reviewable.io/reviews/metabase/metabase/30370#-NUWzmHB-_1DQdD8qc-r:-NUXMu-bCxbV_0jfiow_:b70nl8t) ([raw file](https://github.com/metabase/metabase/blob/99fb47d9e0d149119b3c5e5bd49bf1fd5757be8b/frontend/src/metabase/modes/components/drill/AutomaticInsightsDrill.tsx#L72)):*
<details><summary><i>Previously, ranquild (Alexander Polyankin) wrote…</i></summary><blockquote>

Instead of doing it like this, could we return a single click action with the title of "Automatic insights" and the url of the first drill in this case? So we won't need to call dispatch during the render of the popover
</blockquote></details>

Yes, good point! I re-checked the way how drills availability is checked, and it appeared that they have exactly the same checks. I unified those and removed this part

**Review by ranquild — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/30370)* status: 0 of 15 files reviewed, 4 unresolved discussions (waiting on @deniskaber and @uladzimirdev)

___
*[`frontend/src/metabase/modes/components/drill/AutomaticInsightsDrill.tsx` line 51 at r2](https://reviewable.io/reviews/metabase/metabase/30370#-NUXNVZc3VLkxaszXKnL:-NUXNVZc3VLkxaszXKnM:bbahs6j) ([raw file](https://github.com/metabase/metabase/blob/4b671d4720ae6235f52646fbf04047294fe046b9/frontend/src/metabase/modes/components/drill/AutomaticInsightsDrill.tsx#L51)):*
> ```XML
>   }
> 
>   if (!drillOptions.length) {
> ```

I think now we can simply this - if `automaticInsightsDrill` is false return an empty array

**Pull request merged**